distribution/lib/Standard/Test_New/0.0.0-dev/src/Group.enso

```diff
@@ -9,9 +9,9 @@ type Group_Builder
     Impl builder
 
     ## Specifies a single test.
-    specify : Text -> (Nothing -> Any) -> Nothing
-    specify self (name:Text) ~code =
-        self.builder.append <| Spec.Impl name (_ -> code)
+    specify : Text -> (Nothing | Text) -> (Nothing -> Any) -> Nothing
```

Comment:
Is `(Nothing -> Any)` the right type for a thunk? Isn't it `(-> Any)`?

Reply:
I don't think `(-> Any)` is a valid type at all?

Usually we just don't mark the suspended arguments, and so a suspended thunk is not distinguished in the type signature.

So it would be
```suggestion
    specify : Text -> (Nothing | Text) -> Any -> Nothing
```

I guess we could do better, but that was the convention for now - see e.g. the `<|` operator.

I think there used to be a `Suspended` type - i.e. it could be
```suggestion
    specify : Text -> (Nothing | Text) -> (Suspended Any) -> Nothing
```

It was documented https://github.com/enso-org/enso/blob/develop/docs/semantics/evaluation.md#optional-suspension but I think it has never been used in practice.

Reply:
Specified as `(~code : Any)` and described in the docs on the latest version. And yes, `(-> Any)` does not even compile.

Reply:
`(~code : Any)` looks good!